clang/include/clang/Driver/Options.td

```diff
@@ -4463,49 +4477,50 @@ def Xflang : Separate<["-"], "Xflang">,
 //===----------------------------------------------------------------------===//
 let Flags = [FC1Option, FlangOption, FlangOnlyOption] in {
 
-def cpp : Flag<["-"], "cpp">, Group<f_Group>,
+def cpp : Flag<["-"], "cpp">, Group<gfortran_Group>,
   HelpText<"Enable predefined and command line preprocessor macros">;
-def nocpp : Flag<["-"], "nocpp">, Group<f_Group>,
+def nocpp : Flag<["-"], "nocpp">, Group<gfortran_Group>,
   HelpText<"Disable predefined and command line preprocessor macros">;
 def module_dir : Separate<["-"], "module-dir">, MetaVarName<"<dir>">,
   HelpText<"Put MODULE files in <dir>">,
   DocBrief<[{This option specifies where to put .mod files for compiled modules.
 It is also added to the list of directories to be searched by an USE statement.
 The default is the current directory.}]>;
 
-def ffixed_form : Flag<["-"], "ffixed-form">, Group<f_Group>,
+def ffixed_form : Flag<["-"], "ffixed-form">, Group<fortran_format_Group>,
```

Comment:
Question: What is the difference between `f_Group` and `fortran_format_Group` (or `gfortran_Group`)?

Reply:
f_Group is the original group name that is found in llvm/13.x. Is the Target-independent compilation options
fortran_format_Group is a group for the Fortran source format
gfortran_Group is the gfortran options that we recognize in the driver and pass along when invoking GCC to compile Fortran code.
After initial merge, flang tests fails because it cannot find the symbol, for example, ffixed_form. I don't understand the specifics of each one of it, I based on the code found on release_12x to have an idea how to use the groups names. A more complex merge could be to create a group to encapsulate all three different groups, but as I don't understand specifics of it I cannot tell if that is a good idea and ultimate solution. I vouch for the more simplest solution that was to just change the group name, as it was done between llvm release/12.x and classic release_12x

Reply:
fortran_format_Group is used to help the classic flang driver process the multiple gfortran-style and nvfort/pgfort-style options that control Fortran source format (Mfixed vs -ffixed-form, etc.) I think Abraham's merge here is correct.

In the past, upstream LLVM would have had -ffixed-form in the gfortran_Group to make clang pass it on to gfortran (or ignore it). That functionality broke upstream (I think) and anyway got replaced by new LLVM Flang's support for the options in LLVM 13. LLVM flang places -ffixed-form it in the f_Group (which I think is probably wrong, but that's an upstream bug.) The only implication of that is that the Clang driver (not Flang) will claim all unprocessed f_Group options to prevent an unused argument warning.

Reply:
> The only implication of that is that the Clang driver (not Flang) will claim all unprocessed f_Group options to prevent an unused argument warning.

Thanks @RichBarton-Arm for this explanation. `ToolChains/ClassicFlang.cpp` processes `OPT_fortran_format_Group` but nothing in the code base uses `OPT_gfortran_Group` and `OPT_f_Group`. Could you point me to the Clang driver code that goes through options in `OPT_f_Group` in order to claim them?

Reply:
The bit of code that claims the f_Group options is in Clang's default Driver's ConstructJob function [here](https://github.com/flang-compiler/classic-flang-llvm-project/blob/f4f49d9d0d699f3ac32c1037516c9ab17551991e/clang/lib/Driver/ToolChains/Clang.cpp#L6513)). Seems a shame not to use it to e.g. organise clang's messy --help message to group similar options, but it seems this is all it's used for upstream in clang.

The gfortran support has all completely [bitrotted away](https://reviews.llvm.org/rG6a75496836ea14bcfd2f4b59d35a1cad4ac58cee) upstream in Clang. It will anyway not make much sense when LLVM Flang is  mature. We already agreed that LLVM Flang would have a gfortran-like CLI so we'll want to use these Options.td entries for LLVM Flang options eventually. We might even be able to remove the whole group now. I'll talk to Andrzej about that, but it won't have much bearing on what we're talking about here.

Reply:
Thanks; I didn't realize that `OPT_clang_ignored_f_Group` was related to `OPT_f_Group`. I was trying to understand whether this PR needs to bother changing all the `f_Group` into `gfortran_Group` (for options used by Classic Flang), since it introduces downstream delta that may not really be necessary. If we leave them in `f_Group`, `ToolChains/ClassicFlang.cpp` should claim all of them anyway, and the fact that Clang may try to claim unused copies of such options should be irrelevant.